### PR TITLE
Fetch network using name

### PR DESCRIPTION
**Pull request by ugoamanoh**
#### What's Here

To support retrieving network using either the network name or id, the following was done

1. Created a new private function to iterate through the registered networks for a name match
2. Added the function above the the `Networks.get()` logic
3. New tests to verify searching with name works fine

### CODE REVIEW

**Comment by ugoamanoh** [comment]
@mirceanis I had also considered throwing the `network not configured` error in the `getNetworkByName()` to avoid returning null, but I think it would be better to leave that to the `get()` just in case at any time we decide to;
a. change the execution order of the checks in the `get()`
b. add more checks

File: core/src/main/java/me/uport/sdk/core/Networks.kt [hunk 1]
```diff
@@ -121,13 +121,26 @@ object Networks {
...
+            ?: getNetworkByName(nameOrId)
+            ?: throw IllegalStateException("network [$nameOrId] not configured")
+    }
+
+    /**
+     * Searches for an [EthNetwork] based on a [name]
+     */
+    private fun getNetworkByName(name: String): EthNetwork? {
+        for ((_, value) in NETWORK_CONFIG) {
+            if (value.name === name) {
```
Comment (mirceanis) [inline core/src/main/java/me/uport/sdk/core/Networks.kt]: I think it's worth doing a lowercase comparison here.

File: core/src/test/java/me/uport/sdk/core/NetworksTest.kt [hunk 2]
```diff
@@ -39,6 +39,36 @@ class NetworksTest {
...
+            Networks.get("manet")
+        }.thrownError {
+            isInstanceOf(IllegalStateException::class)
+        }
+    }
+
+    @Test
+    fun `throws error if the network name cases does not match`() {
+        assertThat {
+            Networks.get("Mainnet")
```
Comment (mirceanis) [inline core/src/test/java/me/uport/sdk/core/NetworksTest.kt]: I think this test should actually pass.
Network names (and I think it should also apply to `networkId`s) should not be case-sensitive

File: core/src/main/java/me/uport/sdk/core/Networks.kt [hunk 3]
```diff
@@ -121,13 +121,26 @@ object Networks {
...
-            ?: throw IllegalStateException("network [$networkId] not configured")
+            ?: NETWORK_CONFIG[nameOrId]
+            ?: getNetworkByName(nameOrId)
+            ?: throw IllegalStateException("network [$nameOrId] not configured")
+    }
+
+    /**
+     * Searches for an [EthNetwork] based on a [name]
+     */
+    private fun getNetworkByName(name: String): EthNetwork? {
```
Comment (mirceanis) [inline core/src/main/java/me/uport/sdk/core/Networks.kt]: a shorter version of this method would be:

```kotlin
val queryName = name.toLowerCase()
return NETWORK_CONFIG.values.find { it.name.toLowerCase() == queryName }
```

**Comment by ugoamanoh** [comment]
@mirceanis changes have been made

File: core/src/main/java/me/uport/sdk/core/Networks.kt (hunk 3 above)
Comment (ugoamanoh) [reply core/src/main/java/me/uport/sdk/core/Networks.kt]: This is really good. Thanks

**Pull request merged**